src/core/evaluator.js

```diff
@@ -708,24 +708,39 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
           new ErrorFont('Font ' + fontName + ' is not available'), font));
       }
 
-      var fontRef, xref = this.xref;
+      let fontRef, fontRes, xref = this.xref;
       if (font) { // Loading by ref.
         if (!isRef(font)) {
           throw new Error('The "font" object should be a reference.');
         }
         fontRef = font;
       } else { // Loading by name.
-        var fontRes = resources.get('Font');
+        fontRes = resources.get('Font');
         if (fontRes) {
           fontRef = fontRes.getRaw(fontName);
         } else {
-          warn('fontRes not available');
-          return errorFont();
+          warn(`fontRes "${fontName}" not available.`);
+          // Attempting to fallback, see the code-path just below...
         }
       }
       if (!fontRef) {
-        warn('fontRef not available');
-        return errorFont();
+        if (fontRes) { // Avoid duplicated warning messages.
+          warn(`fontRef "${fontName}" not available.`);
+        }
+        if (!this.options.ignoreErrors && !this.parsingType3Font) {
+          return errorFont();
+        }
+        // Font not found -- sending unsupported feature notification.
+        this.handler.send('UnsupportedFeature',
+                          { featureId: UNSUPPORTED_FEATURES.font, });
+
+        // Falling back to a default font to avoid completely broken rendering,
+        // but note that there're no guarantees that things will look "correct".
+        fontRef = new Dict();
+        fontRef.set('BaseFont', Name.get('PDFJS-FallbackFont'));
+        fontRef.set('Type', Name.get('Type1'));
+        fontRef.set('Subtype', Name.get('Type1'));
```

Comment:
Assuming that it wouldn't cause problems elsewhere, using some kind of dummy Type/Subtype would probably be appropriate since you'd otherwise risk skewing font telemetry results.